tests/bootstrap_tests/test_backups.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+from __future__ import with_statement
+
+import itertools
+import os
+import sys
+from mock import patch
+import sgtk
+from sgtk.util import ShotgunPath
+from sgtk.bootstrap.configuration_writer import ConfigurationWriter
+from shutil import copytree, ignore_patterns
+
+from tank.descriptor import Descriptor, descriptor_uri_to_dict, descriptor_dict_to_uri, create_descriptor, constants
+from tank_test.tank_test_base import setUpModule # noqa
+from tank_test.tank_test_base import temp_env_var
+from tank_test.tank_test_base import TankTestBase
+
+
+class TestBackups(TankTestBase):
+    def setUp(self):
+        super(TestBackups, self).setUp()
+
+        self._tmp_bundle_cache = os.path.join(self.tank_temp, "bundle_cache")
+        self._resolver = sgtk.bootstrap.resolver.ConfigurationResolver(
+            plugin_id="backup_tests",
+            bundle_cache_fallback_paths=[self._tmp_bundle_cache]
+        )
+
+    def test_cleanup(self):
+        """
+        Ensures that after a successful update the backup folder created by the
+        update process is properly deleted 
+        """
+        pathHead, pathTail = os.path.split(__file__)
+        core_path=os.path.join(pathHead,"..", "..")
+        temp_test_path=os.path.join(pathHead, "..", "fixtures", "bootstrap_tests", "test_update")
```

Comment:
A little more detail on the failure @jfboismenu . The reason i added the copy, with call         copytree(core_path, core_copy_path, ignore=ignore_patterns('tests', 'docs')),
is because i get the error above if i attempt to use a tk-core repository with those 2 ignore subfolders. Either way if i make the copy with those 2 subfolders or if i attempt to use tk-core in core_path variable directly, the fact that those 2 folders are in there will trip up the update_configuration/create_tank_command() functions you see in the error output. I ran into some code where it seemed like the app store tk-core was being stripped of those 2 folders, so i did the same thing, and then my those functions stopped being tripped up.

I feel like I may be missing something from your suggestion. Let's get in sync and discuss this further. But this was the only way i was able to get tk-core to be installed in the appropriate location